src/plugins/EnergyRAPL.cpp

```diff
@@ -0,0 +1,154 @@
+#include "EnergyRAPL.h"
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+#ifdef ENABLE_MPI
+#include <map>
+#endif
+
+#include "../utils/Logger.h"
+#include "utils/xmlfileUnits.h"
+
+EnergyRAPL::RAPLCounter::RAPLCounter(const std::string domainBasePath) {
+	// File path for reading current micro joules
+	std::ostringstream microJoulesPath;
+	microJoulesPath << domainBasePath << "/energy_uj";
+	_microJoulesPath = microJoulesPath.str();
+	// Range, i.e., maximum value of RAPL energy counter, in micro-joules
+	std::ostringstream rangeMicroJoulesPath;
+	rangeMicroJoulesPath << domainBasePath << "/max_energy_range_uj";
+	std::ifstream rangeMicroJoulesFile(rangeMicroJoulesPath.str());
+	rangeMicroJoulesFile >> _rangeMicroJoules;
+	reset();
+}
+
+void EnergyRAPL::RAPLCounter::reset() {
+	// Update last micro joules
+	update();
+	_microJoules = 0;
+}
+
+double EnergyRAPL::RAPLCounter::update() {
+	long long currentMicroJoules;
+	std::ifstream packageIdFile(_microJoulesPath);
+	packageIdFile >> currentMicroJoules;
+	long long deltaMicroJoules = currentMicroJoules - _lastMicroJoules;
+	// Correct counter overflow (occurs around every 60 seconds)
+	if (0 > deltaMicroJoules) {
+		deltaMicroJoules += _rangeMicroJoules;
+	}
+	_lastMicroJoules = currentMicroJoules;
+	_microJoules += deltaMicroJoules;
+	return ((double)_microJoules) / 1000 / 1000;
```

Comment:
```suggestion
	return static_cast<double>(_microJoules) * 1e-6;
```

1. Use `double` constants when you want `double`s.
1. Why to divisions when you can also do one multiplication?
1. No C-Style casts
1. Please add a comment why `1e-6` :)